internal/ignore/path_matcher.go

```diff
@@ -119,3 +123,12 @@ func (m tempBrokenSymlinkMatcher) Matches(path string, isDir bool) (bool, error)
 
 	return ospath.IsBrokenSymlink(path)
 }
+
+type directoryMatcher struct {
+	dir string
+}
+
+func (d directoryMatcher) Matches(p string, isDir bool) (bool, error) {
```

Comment:
dumb Q: will this match itself?

Reply:
It turns out that yes it will, but I consider that a bug in `ospath.Child` since a directory is pretty definitely not a child of itself.

Either way, I don't think whether it will or not is particularly important for this feature.